### PR TITLE
[1LP][RFR] Add 2 volume backup restore tests

### PR DESCRIPTION
**Pull request by jadhaj**
Purpose or Intent
=================
Add tests: 
test_restore_volume_backup
test_restore_incremental_backup

{{ pytest: cfme/tests/openstack/cloud/test_volume_backup.py::test_restore_volume_backup cfme/tests/openstack/cloud/test_volume_backup.py::test_restore_incremental_backup }}

### CODE REVIEW

**Comment by jadhaj** [comment]
Local result:

2018-10-07 03:07:43,852 [I] [cfme] ----------- test_restore_volume_backup[openstack-11] result: passed ------------ (cfme/fixtures/log.py:70)
2018-10-07 03:50:27,735 [I] [cfme] --------- test_restore_incremental_backup[openstack-11] result: passed --------- (cfme/fixtures/log.py:70)
(cfme_

**Comment by digitronik** [comment]
PRT showing creates failures can you please debug those?

**Comment by jadhaj** [comment]
@digitronik : The tests are passing on my env therefore I can't debug them locally.
That's why embed local results on PR

**Comment by jadhaj** [comment]
@digitronik I got your point. please let me know if below steps are enough:
1. On incremental_backup fixture, add if volume.status == 'available': check before backup creation
2. On  test_restore_volume_backup and test_restore_incremental_backup tests, add if volume_backup.status == 'available': check before restore

**Comment by digitronik** [comment]
PRT error is in setup for tests. which is not related to the actual test. Connection problem with `wharf`. Moving to 1LP.

File: cfme/tests/openstack/cloud/test_volume_backup.py [hunk 1]
```diff
@@ -53,10 +53,11 @@ def incremental_backup(volume_backup, provider):
...
     # create incremental backup for a volume with existing backup
-    backup_name = fauxfactory.gen_alpha()
-    volume.create_backup(backup_name, incremental=True)
-    incremental_backup = backup_collection.instantiate(backup_name, provider)
-    yield incremental_backup
+    if volume.status == 'available':
+        backup_name = fauxfactory.gen_alpha()
+        volume.create_backup(backup_name, incremental=True)
+        incremental_backup = backup_collection.instantiate(backup_name, provider)
+        yield incremental_backup
```
Comment (mshriver) [inline cfme/tests/openstack/cloud/test_volume_backup.py]: In the case where volume.status is not available, this won't yield or return, and L63 will result in an UnboundLocal when `incremental_backup` was never created.

There needs to be an else clause here with a `pytest.skip` call. :)

Comment (mshriver) [reply cfme/tests/openstack/cloud/test_volume_backup.py]: Good change, still seeing a test failure in PRT with:

```
>               raise AssertionError('assert_no_error: {}: {}'.format(message.type, message.text))
E               AssertionError: assert_no_error: error: Unable to create backup for Cloud Volume "PurJWHUPoY": Invalid backup: The parent backup must be available for incremental backup.

/usr/lib/python2.7/site-packages/widgetastic_patternfly/__init__.py:228: AssertionError
AssertionError
assert_no_error: error: Unable to create backup for Cloud Volume "PurJWHUPoY": Invalid backup: The parent backup must be available for incremental backup.
```

Comment (jadhaj) [reply cfme/tests/openstack/cloud/test_volume_backup.py]: The point is that locally both tests are passing. what would you suggest me to do further

**Comment by izapolsk** [comment]
your commit isn't signed

**Comment by izapolsk** [comment]
I've my added empty signed commit in order to start PRT.

**Comment by izapolsk** [comment]
1. PRT failed because volume is full. @jadhaj please push @ pakotvan to clean up openstack provider.
2. @jadhaj please fix lint issue in Travis.

**Comment by jadhaj** [comment]
@izapolsk Sorry but what is pakotvan ?

**Pull request merged**